packages/airnode-node/src/workers/local-gateways/validation.ts

```diff
@@ -123,16 +123,19 @@ const beaconSchema = z.object({
   airnodeAddress: z.string(),
   endpointId: z.string(),
   encodedParameters: z.string(),
-  // Fields encodedValue, timestamp and signature are optional as they might be missing if we have no data for some of the beacons.
-  // This is fine as we would ignore those but we still need to know how many beacons are in the beacon set and Airnode
-  // addresses to derive the data feed ID.
-  encodedValue: z.string().optional(),
-  timestamp: z.string().optional(),
-  signature: z.string().optional(),
+  // Signed data might for some of the beacons (as long as the majority has the data). We still need to know all of the
+  // beacons to derive the data feed ID.
+  signedData: z
+    .object({
+      encodedValue: z.string(),
+      timestamp: z.string(),
+      signature: z.string(),
+    })
+    .optional(),
```

Comment:
The difference is that signed data is optional, but if present it must have all of the fields defined.

Reply:
I understand the difference but all my arguments still apply.

Reply:
I disagree it doesn't bring anything. This way is less conceptually cleaner and ensures you get fill signed data or nothing. I will rather update the other pieces and have this implemented properly, while it's not yet used by users.

Reply:
This is not a public endpoint and it will not be used by users. And I understand your point and I agree that it might be conceptually cleaner. I'm just saying that the time writing it, updating all the other parts I mentioned, and arguing about it is not worth the "cleaner" solution.
The only practical difference was a change from
```ts
  const beaconsWithData = beacons.filter((beacon) => beacon.encodedValue && beacon.timestamp && beacon.signature);
```
to
```ts
  const beaconsWithData = beacons.filter((beacon) => beacon.signedData) as Required<Beacon>[];
```
and that's it.